cohttp-eio/src/server.ml

```diff
@@ -3,44 +3,42 @@ module IO = Io.IO
 
 type body = Body.t
 type conn = IO.conn * Cohttp.Connection.t [@@warning "-3"]
+type writer = IO.oc
+type response = writer -> unit
 
 type response_action =
-  [ `Expert of Http.Response.t * (IO.ic -> IO.oc -> unit IO.t)
-  | `Response of Http.Response.t * body ]
-
-(* type handler =
- *   sw:Eio.Switch.t ->
- *   Eio.Net.Sockaddr.stream ->
- *   Http.Request.t ->
- *   Eio.Flow.source ->
- *   Http.Response.t * Eio.Flow.source *)
+  [ `Expert of Http.Response.t * (IO.ic -> IO.oc -> unit)
+  | `Response of response ]
 
 type t = {
   conn_closed : conn -> unit;
-  handler : conn -> Http.Request.t -> body -> response_action IO.t;
+  handler : conn -> Http.Request.t -> body -> IO.ic -> IO.oc -> unit;
```

Comment:
My only concern is that the change to old-style response was done to align the APIs, with this they APIs are diverging again (at least in the expectations and signatures)

Reply:
Yes, although not when using the normal response functions. So it's certainly possible to write code that works with both.